analyzers/tests/SonarAnalyzer.Test/Rules/SpecifyRouteAttributeTest.cs

```diff
@@ -0,0 +1,92 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#if NET
+
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class SpecifyRouteAttributeTest
+{
+    private readonly VerifierBuilder builder = new VerifierBuilder<SpecifyRouteAttribute>()
+        .WithOptions(ParseOptionsHelper.FromCSharp12)
+        .AddReferences([
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcCore,
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreHttpAbstractions,
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcViewFeatures,
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcAbstractions,
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreHttpFeatures,
+        ]);
+
+    [TestMethod]
+    public void SpecifyRouteAttribute_CS() =>
+        builder.AddPaths("SpecifyRouteAttribute.cs").Verify();
+
+    [TestMethod]
+    public void SpecifyRouteAttribute_PartialClasses_CS() =>
+        builder
+            .AddSnippet("""
+                        using Microsoft.AspNetCore.Mvc;
+                        using Microsoft.AspNetCore.Mvc.Routing;
+
+                        public partial class HomeController : Controller       // Noncompliant
+                        {
+                            [HttpGet("Test")]
+                            public IActionResult Index() => View();            // Secondary
+                        }
+                        """)
+            .AddSnippet("""
+                        using Microsoft.AspNetCore.Mvc;
+                        using Microsoft.AspNetCore.Mvc.Routing;
+
+                        public partial class HomeController : Controller       // Noncompliant
+                        {
+                        }
+                        """)
+            .Verify();
+
+    [TestMethod]
+    public void SpecifyRouteAttribute_PartialClasses_OneGenerated_CS() =>
+        builder
+            .AddSnippet("""
+                        using Microsoft.AspNetCore.Mvc;
+                        using Microsoft.AspNetCore.Mvc.Routing;
+
+                        public partial class HomeController : Controller       // Noncompliant
+                        {
+                            [HttpGet("Test")]
+                            public IActionResult Index() => View();            // Secondary
+                        }
+                        """)
+            .AddSnippet("""
+                        // <auto-generated/>
+                        using Microsoft.AspNetCore.Mvc;
+                        using Microsoft.AspNetCore.Mvc.Routing;
+
+                        public partial class HomeController : Controller
+                        {
```

Comment:
Add a non-compliant action here. I'm not sure if secondary locations are filtered or not. I'm also not sure if reporting a secondary location in a generated file is a problem or not. But it would be interesting to learn.

Reply:
Good point. These are not filtered.

Reply:
I will not change the behavior, I will just document it.